zerver/tests/test_email_notifications.py

```diff
@@ -1546,6 +1546,20 @@ def test_multiple_stream_messages_and_mentions(self) -> None:
         email_subject = "#Denmark > test"
         self.assertEqual(mail.outbox[0].subject, email_subject)
 
+    def test_latex_math_formulas(self) -> None:
+        hamlet = self.example_user("hamlet")
+        msg_id = self.send_stream_message(
+            self.example_user("iago"), "Denmark", "Equation: $$d^* = +\\infty$$"
+        )
+        handle_missedmessage_emails(
+            hamlet.id,
+            [
+                {"message_id": msg_id, "trigger": "stream_email_notify"},
+            ],
+        )
+        self.assert_length(mail.outbox, 1)
+        self.assertIn("Equation: $$d^* = +\\infty$$", mail.outbox[0].message().as_string())
+
```

Comment:
- I couldn't figure out how to test the html version of the email's content.
- While I was able to view the email notifications in `/emails` but I don't know where to look for the mobile push notifications.

Reply:
@mateuszmandera maybe you can help give advice on the testing questions here? I don't know off the top of my head.

Reply:
@akshatdalton For email html content in an automated test, see how this code grabs it for example, I think that should work:
```
class EmailLogBackEnd(EmailBackend):
    @staticmethod
    def log_email(email: EmailMessage) -> None:
        """Used in development to record sent emails in a nice HTML log"""
        html_message: Union[bytes, EmailMessage, Message, str] = "Missing HTML message"
        assert isinstance(email, EmailMultiAlternatives)
        if len(email.alternatives) > 0:
            html_message = email.alternatives[0][0]
```

> I don't know where to look for the mobile push notifications.

While I'm not aware of an ergonomic way of testing this manually, what I'd typically do here is just modify the `consume` code in `PushNotificationsWorker` to `print` the even dict to the console and verify it looks like it should. Or are you asking about writing an automated test? For that you could mock the `handle_push_notification` function there to assert it's being with appropriate arguments